consensus/consensus.go

```diff
@@ -124,6 +124,9 @@ type Engine interface {
 	// CreateSnapshot does not return a snapshot but creates a new snapshot if not exists at a given point in time.
 	CreateSnapshot(chain ChainReader, number uint64, hash common.Hash, parents []*types.Header) error
 
+	// GetHeadersToApply returns the headers need to be applied to calculate snapshot for the given block number.
+	GetHeadersToApply(chain ChainReader, number uint64, hash common.Hash, parents []*types.Header) ([]*types.Header, error)
```

Comment:
Let's make it super specific because this function is used for very specific case. `GetKaiaHeadersForSnapshotAt`?

Reply:
Applied at [8f04373](https://github.com/kaiachain/kaia/pull/48/commits/8f043730bf82ae3e1211cbf37988c2e20ee35b76)